CMakeLists.txt

```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
 add_subdirectory(3rdparty/software)
 add_subdirectory(3rdparty/plutovg)
 
-if(BUILD_SHARED_LIBS)
-    target_compile_definitions(lunasvg PUBLIC LUNASVG_SHARED)
-    target_compile_definitions(lunasvg PRIVATE LUNASVG_EXPORT)
+set_target_properties(lunasvg
+    PROPERTIES
+    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+)
+
+add_library(lunasvg::lunasvg ALIAS lunasvg)
+
+if(NOT BUILD_SHARED_LIBS)
+    set_target_properties(lunasvg
```

Comment:
Usually I see target_compile_definitions() for this, not sure whether it makes a difference

Reply:
CMake recommends using definitions here: https://cmake.org/cmake/help/latest/command/target_compile_options.html#see-also

Reply:
Done, thanks.